### PR TITLE
nyxt: add module

### PR DESCRIPTION
**Pull request by theesm**
### Description

Added module for [Nyxt browser: The hacker's browser](https://nyxt.atlas.engineer/). This MR is still WIP as tests still have to be written and comaptibility to Non-NixOS-Systems has to be evaluated.

### Checklist

- [X] Change is backwards compatible.

- [X] Code formatted with `./format`.

- [X] Code tested through `nix-shell --pure tests -A run.all`.

- [X] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [X] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [X] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [X] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: .github/CODEOWNERS
Comment (ncfavier) [inline .github/CODEOWNERS:1]: This file was removed today.

File: modules/programs/nyxt.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  options.programs.nyxt = {
+    enable = mkEnableOption "Nyxt";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.nyxt;
+      defaultText = literalExpression "pkgs.nyxt";
+      description = "The Nyxt package to install";
+    };
```
Comment (ncfavier) [inline modules/programs/nyxt.nix:17]:
```suggestion
    package = mkPackageOption pkgs "nyxt" { };
```

File: .github/CODEOWNERS
Comment (theesm) [reply .github/CODEOWNERS:1]: Did that to resolve said conflicts. Also added tests.

**Comment by theesm** [comment]
I’m not using NixOS/home-manager anymore, so closing this seems the right thing to do.

**Pull request closed without merging**